odc/geo/_xr_interop.py

```diff
@@ -245,6 +246,96 @@ def assign_crs(
     return xx
 
 
+def mask(xx: XrT, poly: Geometry, all_touched: bool = True) -> XrT:
+    """
+    Apply a polygon geometry as a mask, setting all xr.Dataset
+    or xr.DataArray pixels outside the rasterized polygon to ``NaN``.
+
+    :param xx:
+       :py:class:`~xarray.Dataset` or :py:class:`~xarray.DataArray`.
+
+    :param poly:
+       Geometry shape used to mask ``xx``.
+
+    :param all_touched:
+        If ``True``, all pixels touched by ``poly`` will remain unmasked.
+        If ``False``, only pixels whose center is within the polygon or
+        that are selected by Bresenham's line algorithm will remain unmasked.
+
+    :return:
+        A :py:class:`~xarray.Dataset` or :py:class:`~xarray.DataArray`
+        masked by ``poly``.
+    """
+
+    # Reproject `poly` to match `xx`
+    poly = poly.to_crs(crs=xx.odc.crs)
```

Comment:
This is done internally by `rasterize`, so it's best not to repeat it here. Change in coordinate system for a geometry is often a "dangerous" thing that can break geometry. While `rasterize` doesn't currently do anything different than this code, it's best to keep number of places that perform such conversion to a minimum.

What `rasterize` **should** do when given a geometry in a projection other than what's desired on output

1. Call `.to_crs` with appropriate resolution computed from mask resolution
2. Check if any nodes failed to map from source to destination CRS, and if so, try again with a more costly approach which is not currently implemented, a kind of `robust_and_costly_to_crs(geom, crs, clip_region)`.

Reply:
Cool, I didn't realise it was already handled in `rasterize`. Removed and everything appears to still work nicely.